honeybee_grasshopper_core/src/HB Intersect Solids.py

```diff
@@ -0,0 +1,125 @@
+# Honeybee: A Plugin for Environmental Analysis (GPL)
+# This file is part of Honeybee.
+#
+# Copyright (c) 2019, Ladybug Tools.
+# You should have received a copy of the GNU General Public License
+# along with Honeybee; If not, see <http://www.gnu.org/licenses/>.
+# 
+# @license GPL-3.0+ <http://spdx.org/licenses/GPL-3.0+>
+
+
+"""
+Take a list of closed breps (polysurfaces) that you intend to turn into Rooms and
+split their Faces to ensure that there are matching faces between each of the
+adjacent rooms.
+_
+Matching faces and face areas betweem adjacent rooms are necessary to ensure
+that the conductive heat flow calculation occurs correctly across the face in
+an energy simulation.
+-
+
+    Args:
+        _solids: A list of closed Rhino breps (polysurfaces) that you intend to turn
+            into Rooms that do not have perfectly matching surfaces between
+            adjacent Faces (this matching is needed to contruct a correct
+            multi-room energy model).
+        _run: Set to True to run the component.
+    
+    Returns:
+        int_solids: The same input closed breps that have had their component
+            faces split by adjacent polysurfaces to have matching surfaces between
+            adjacent breps.  It is recommended that you bake this output and check
+            it in Rhino before turning the breps into honeybee Rooms.
+"""
+
+ghenv.Component.Name = "HB Intersect Solids"
+ghenv.Component.NickName = 'IntSolid'
+ghenv.Component.Message = '0.1.0'
+ghenv.Component.Category = "HoneybeeCore"
+ghenv.Component.SubCategory = '0 :: Create'
+ghenv.Component.AdditionalHelpFromDocStrings = "2"
+
+
+###############################################################################
+#########################     NOTE TO DEVELOPERS    ###########################
+###############################################################################
+"""
+The code within this component will be replaced with a much more efficient
```

Comment:
We already had this discussion and we discovered that the native Grasshopper components don't always do what we want (several common cases fail with those native GH components).  

Also, perhaps I should have been clearer in that note to developers that, after seeing how OpenStudio SDK does this type of intersection and realizing some important techniques during my development of ladybug_geometry, I am confident that I can write this component to be a couple orders of magnitude faster (certainly much faster than the native GH components). Simple things like bounding box checks (which OpenStudio SDK uses) and co-planartiy checks (which I use throughout ladybug_geometry) should get this component to run in less than 1% of the time it does now for large models.

I really wanted to implement this all as part of this first PR since it's really an important feature for certain large urban models. Let me see if I can get the honeybee_energy stuff to a good minimal state before we merge this in and I will add it.

Reply:
As you wish!